### PR TITLE
BD-2606: Movable Ink Customer Data API docs

### PR DESCRIPTION
**Pull request by bre-fitzgerald**
# Pull Request/Issue Resolution

#### Description of Change:
Adds a new article for **Connecting to the Customer Data API** (name can be changed), which covers the prerequisites, integration steps, considerations, and sample payloads provided in the documents from Movable Ink. 

Closes #**BD-2606**

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [x] No

### CODE REVIEW

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+{% endtab %}
+{% tab Example webhook %}
+
+{% raw %}
+
+```liquid
+
+```
+
+{% endraw %}
```
Comment (bre-fitzgerald) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:349]: Document was missing an example webhook for the "Card Add" event

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
```
Comment (bre-fitzgerald) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:389]: Document was missing an example webhook for the "Product Search" event

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 3]
```diff
@@ -262,13 +302,41 @@ In this example, a hashed email address is used as the `anonymous_id` for users
 
 ### Cart Add Event
```
Comment (alex-braze) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:303]: I think we can remove the tabs for events that don't have an example webhook: Cart Add Event, Product Search Event and Conversion Event (this one doesn't have a webhook campaign example only a /users/track payload example so we can remove it)

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 4]
```diff
@@ -331,19 +496,46 @@ In this example, a hashed email address is used as the `anonymous_id` for users
 
 ### Identify Event
```
Comment (alex-braze) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:497]: Similar to the above can we change the tab to be "Expected MovableInk Request Payload" and can we add a tab to the left that is for the "Example Braze Trigger Event" with an example of what the Braze event payload would look like for an Identify Event that is triggering the webhook campaign to MI:

`{
  "events": [
    {
      "external_id": "123456789",
      "name": "Account Created",
      "time": "2023-12-06T19:20:45+01:00"
    }  
  ]
}`

**Review by alex-braze — COMMENTED** [review]
@bre-fitzgerald added some comments in with some payloads!

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md (hunk 3 above)
Comment (bre-fitzgerald) [reply _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:361]: Updated!

**Comment by bre-fitzgerald** [comment]
@alex-braze I believe I resolved all of your comments; let me know if this looks okay to merge or if any additional changes or review is needed. Thanks!

**Comment by alex-braze** [comment]
@bre-fitzgerald apologies, one last comment: I think we can remove the "Cart Add Event", "Product Search Event" and "Conversion Event" example payloads completely!

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+
+## Considerations
+
+### Aligning on a unique user identifier
+
+Make sure the unique user identifier (UUID) value that you're using as your `mi_u`, is available within Braze and can be included in the event payloads sent to Movable Ink.
+
+This ensures that the behavioral events Movable Ink references when generating an image are associated with the same customer they received the behavioral events for. If the UUID value is not the same as the Braze `external_id`, the UUID must be captured and passed to Braze as an attribute or in the event properties of a Braze event to leverage this identifier.
+
+Braze tracks user behavior across multiple platforms (web, mobile app, etc.), so a single user may have several distinct anonymous IDs. These IDs can be merged into the single known Stories user profile when an `identify` event is sent to Movable Ink, as long as the `identify` event includes both an anonymous identifier and the single known identifier.
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:126]:
```suggestion
Braze tracks user behavior across multiple platforms (such as web and mobile app), so a single user may have several distinct anonymous IDs. These IDs can be merged into the single known Stories user profile when an `identify` event is sent to Movable Ink, as long as the `identify` event includes both an anonymous identifier and the single known identifier.
```

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+
+Make sure not to use A/B testing in your campaign by clearing the **Control Group** checkbox. If a control group is included, a percentage of users will not have data sent to Movable Ink. All of your audience should go to the variant rather than the control group.
+
+![A/B Testing panel in a Braze campaign with 100% variant distrubtion assigned to Variant 1, and no control group.][img5]
+
+#### Step 2c: Choose conversion events (optional)
+
+If desired, you can assign conversion events to this campaign within Braze.
+
+However, given that the webhook is intended just to stream data, attribution at this level is likely less useful than looking at attribution at the campaign level after the behavioral data from Braze is used to personalize content.
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:112]:
```suggestion
However, given that the webhook is intended to stream data, attribution at this level is likely less useful than looking at attribution at the campaign level after the behavioral data from Braze is used to personalize content.
```

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+
+![Webhook response message in Braze showing a 200 OK response.][img4]{: style="max-width:75%" }
+
+### Step 2: Finalize your campaign setup
+
+#### Step 2a: Schedule your campaign
+
+When you're done composing and testing the webhook, [schedule your campaign][2]. 
+
+Braze supports scheduled, action-based, and API-triggered deliveries. [Action-based delivery][3] is usually the best fit for most behavioral event use cases. However, if there are any questions about which may make sense for your use case, discuss with your Braze and Movable Ink customer success managers.
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:88]:
```suggestion
Braze supports scheduled, action-based, and API-triggered deliveries. [Action-based delivery][3] is usually the best fit for most behavioral event use cases. For questions about what makes sense for your use case, connect with your Braze and Movable Ink customer success managers.
```

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+      }
+    }
+  ]
+}
+```
+
+{% endraw %}
+
+{% endtab %}
+{% tab Expected MovableInk Request Payload %}
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:193]:
```suggestion
{% tab Expected Movable Ink Request Payload %}
```
Not sure if this is supposed to be two words

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+      "name": "Account Created",
+      "time": "2023-12-06T19:20:45+01:00"
+    }
+  ]
+}
+```
+
+{% endraw %}
+{% endtab %}
+{% tab Expected MovableInk Request Payload %}
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:382]:
```suggestion
{% tab Expected Movable Ink Request Payload %}
```
Same comment

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+      }
+    }
+  ]
+}
+```
+
+{% endraw %}
+
+{% endtab %}
+{% tab Expected MovableInk Request Payload %}
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:305]:
```suggestion
{% tab Expected Movable Ink Request Payload %}
```
Same comment

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+
+#### Forwarding Braze events for anonymous and known users
+
+If you want to forward Braze events from anonymous users (users before an `external_id` is assigned to their profile), you'll need to decide which identifier to use as the `anonymous_id` for Movable Ink until an `external_id` becomes available. Choose an `anonymous_id` that will remain constant on your Braze user profile. You can use Liquid logic in the webhook body to decide whether to pass an `anonymous_id` or a `user_id`.
+
+For more, refer to the example webhooks under [sample payloads](#sample-payloads).
+
+## Sample payloads
+
+### Product View Event
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:156]:
```suggestion
### Product view event
```
?

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+  "type": "track",
+}
+
+```
+
+{% endraw %}
+{% endtab %}
+{% endtabs %}
+
+### Category View Event
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:278]:
```suggestion
### Category view event
```
Feel free to disregard if inaccurate

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 13]
```diff
@@ -0,0 +1,440 @@
...
+}
+
+```
+
+{% endraw %}
+
+{% endtab %}
+{% endtabs %}
+
+### Identify Event
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:361]:
```suggestion
### Identify event
```

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md [hunk 14]
```diff
@@ -0,0 +1,440 @@
...
+| Content-Type | application/json |
+| Authorization | Enter the Basic Authentication you received from Movable Ink. |
+{: .reset-td-br-1 .reset-td-br-2}
+
+![Settings tab of the webhook composer in Braze with key-value pairs for Content-Type and Authorization.][img2]{: style="max-width:75%" }
+
+#### Step 1c: Configure your payload
+
+1. Return to the **Compose** tab.
+2. For your **Request Body**, either create your own request body with JSON key-value pairs, or enter your event payload as raw text. Refer to the [sample payloads](#sample-payloads) for examples of standard eCommerce events.
```
Comment (lydia-xie) [inline _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:62]:
```suggestion
2. For your **Request Body**, either create your own request body with JSON key-value pairs or enter your event payload as raw text. Refer to the [sample payloads](#sample-payloads) for examples of standard ecommerce events.
```

**Review by lydia-xie — CHANGES_REQUESTED** [review]
made some suggestions for your review! looks great overall

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md (hunk 6 above)
Comment (bre-fitzgerald) [reply _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:112]: revised your suggestion slightly

File: _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md (hunk 8 above)
Comment (bre-fitzgerald) [reply _docs/_partners/message_personalization/dynamic_content/movable_ink/customer_data_api.md:193]: oop good catch, copied and pasted from another source and missed this 👍

**Review by lydia-xie — APPROVED** [review]

**Pull request merged**